core/src/main/java/org/infinispan/configuration/cache/CacheMode.java

```diff
@@ -5,6 +5,11 @@
  */
 public enum CacheMode {
    /**
+    * Data is not replicated and certain features are not possible
```

Comment:
I had XML configuration in mind - the checks against prohibited configuration option are needed for programmatic configuration anyway.
And I agree that I could elaborate about the features a bit more here, though the documentation of other modes is not verbose neither.